1348 transfer dark mode images components

GCHQ-Developer-299 opened the pull request:
<!-- 🙏 Thank you for your contribution, it is greatly appreciated! -->
<!-- Please check our Contributing Guidance https://github.com/mi6/ic-design-system/blob/develop/CONTRIBUTING.md before creating a PR. -->

## Summary of the changes

Added dark mode images for every component page.
Divider has had the Appearance section of its guidance page removed as it's not applicable to v3

Animations and recordings have not been updated (partly because they don't play with the index.tsx importing, partly due to time pressure. This will need to be completed as a separate ticket. 

## Related issue

#1348

GCHQ-Developer-741 commented:
I'm seeing a regression from v3 develop regarding the vertical card icon
![image](https://github.com/user-attachments/assets/219f972a-e4a4-4650-9718-a57129649b8f)

V3 develop: 
![image](https://github.com/user-attachments/assets/4e5edfe8-b4e9-48ce-9690-100c9fd85d97)

Are you seeing this locally?

GCHQ-Developer-741 commented:
These images are the wrong way round
![image](https://github.com/user-attachments/assets/ff5d8609-a89f-4381-9faf-81a50e569a19)

GCHQ-Developer-299 commented:
> These images are the wrong way round ![image](https://private-user-images.githubusercontent.com/117368893/418960068-ff5d8609-a89f-4381-9faf-81a50e569a19.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEwODUzNDgsIm5iZiI6MTc0MTA4NTA0OCwicGF0aCI6Ii8xMTczNjg4OTMvNDE4OTYwMDY4LWZmNWQ4NjA5LWE4OWYtNDM4MS05ZmFmLTgxYTUwZTU2OWExOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMwNFQxMDQ0MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTFiMzc3NGE3MzU2ZTY5NTIwZWQyNThjOWUyNDYyNjk3MDVlZTViNmU4ODU3ZjQyNmRjZDJhMDIwNmE1Nzc4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RKMoEZr7aSHhqxPnsCmFfjvBiArELo36hu9ry8JPJEA)

Whoops! The figures are numbered in the wrong sequence, which tripped me up (Fig4 goes before Fig3...)  I've now fixed :) 

Regards the Card regression - I am seeing it, but no idea why. I've rebased off v3.0.0, hopefully that resolves the issue? I don't think any of my changes have affected card styles!

GCHQ-Developer-741 commented:
There are one or two guidance adjustments that could be made:

1) [Toggle button](https://mi6.github.io/ic-design-system-githubpages/branches/1348-transfer-dark-mode-images-components/components/toggle-button#colour) - The appearance prop is mentioned under the colour header.

2) [Link](https://mi6.github.io/ic-design-system-githubpages/branches/1348-transfer-dark-mode-images-components/components/link/#colour)- _"If the text appears on a different coloured background, use the light or dark theme links with monochrome={true} to achieve good colour contrast."_ could be removed, along with the second sentence on the dont example (_"Only use the same theme if the text is used on a dark background."_). This bit is more appropriate for the example immediately after this, I would argue. In this example (specifically when viewed in dark mode) it looks like we are contradicting ourselves by showing a monochrome link on a dark background in a don't window.
![image](https://github.com/user-attachments/assets/49f16f56-2d48-4bb4-b4a0-b4ae2b3d1c61)
 I think overall the meaning of this guidance is better shown in the below screenshot, and should be removed from the former to avoid confusion.
![image](https://github.com/user-attachments/assets/bf53f7de-13a6-4b46-af20-276ee0ad92d7)

I hope I've explained that clearly.

GCHQ-Developer-299 commented:
> There are one or two guidance adjustments that could be made:
> 
> 1. [Toggle button](https://mi6.github.io/ic-design-system-githubpages/branches/1348-transfer-dark-mode-images-components/components/toggle-button#colour) - The appearance prop is mentioned under the colour header.
> 2. [Link](https://mi6.github.io/ic-design-system-githubpages/branches/1348-transfer-dark-mode-images-components/components/link/#colour)- _"If the text appears on a different coloured background, use the light or dark theme links with monochrome={true} to achieve good colour contrast."_ could be removed, along with the second sentence on the dont example (_"Only use the same theme if the text is used on a dark background."_). This bit is more appropriate for the example immediately after this, I would argue. In this example (specifically when viewed in dark mode) it looks like we are contradicting ourselves by showing a monochrome link on a dark background in a don't window.
>    ![image](https://private-user-images.githubusercontent.com/117368893/419026244-49f16f56-2d48-4bb4-b4a0-b4ae2b3d1c61.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEwOTg1ODksIm5iZiI6MTc0MTA5ODI4OSwicGF0aCI6Ii8xMTczNjg4OTMvNDE5MDI2MjQ0LTQ5ZjE2ZjU2LTJkNDgtNGJiNC1iNGEwLWI0YWUyYjNkMWM2MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMwNFQxNDI0NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZDlkNDc4ZmRiYjdjZjJmZmViMTg5NThmNzAyNjY3NWNmMzM1MGI0YjI2OGZiNjQ5ZTBlZTIwNmMxMjk4YzM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.u7iAGbuayLsDcYK-hbExPPDsnQUnWTeQDW57Z8Xt9fc)
>    I think overall the meaning of this guidance is better shown in the below screenshot, and should be removed from the former to avoid confusion.
>    ![image](https://private-user-images.githubusercontent.com/117368893/419026579-bf53f7de-13a6-4b46-af20-276ee0ad92d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEwOTg1ODksIm5iZiI6MTc0MTA5ODI4OSwicGF0aCI6Ii8xMTczNjg4OTMvNDE5MDI2NTc5LWJmNTNmN2RlLTEzYTYtNGI0Ni1hZjIwLTI3NmVlMGFkOTJkNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMwNFQxNDI0NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOWE2NjJkMzE5M2M5ZTkxOTc4MzRlOWMyY2NlNzUwZjhmYWFmZDRlNDJhNjFhNzQxNzhhNTdkM2Q0MjFkNjQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.NELWqyjkhs1OFLaaAAhZTTF0FgxWHXlyon4RxHkU-S4)
> 
> I hope I've explained that clearly.

Addressed your comment in an additional commit :)
